src/main/java/com/blog/domain/post/controller/dto/response/PostListResponse.java

```diff
@@ -0,0 +1,53 @@
+package com.blog.domain.post.controller.dto.response;
+
+import com.blog.domain.post.controller.dto.request.PostContentDto;
+import com.blog.domain.post.domain.ContentType;
+import com.blog.domain.post.domain.Post;
+import com.blog.domain.post.domain.PostContent;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public record PostListResponse(
+        Long postId,
+        String title,
+        List<PostContentDto> postContents,
+        int commentCount,
+        String imageUrl,
+        LocalDate createdAt
+) {
+
+    public static PostListResponse from(Post post, List<PostContent> postContents) {
+        List<PostContentDto> postContentDtos = toPostContentDtos(postContents);
+        String thumbnailUrl = extractThumbnailUrl(postContentDtos);
+        return new PostListResponse(
+                post.getId(),
+                post.getTitle(),
+                postContentDtos,
+                post.getCommentCount(),
+                thumbnailUrl,
+                post.getCreatedAt()
+        );
+    }
+
+    // postContents에서 postContentDto로 변환
+    private static List<PostContentDto> toPostContentDtos(List<PostContent> postContents) {
+        return postContents.stream()
+                .map(content -> new PostContentDto(
+                        ContentType.valueOf(content.getType().name()),
+                        content.getContent()
+                ))
+                .collect(Collectors.toList());
+    }
+
```

Comment:
아마 PR에서 말씀해주신 부분이 이 부분인것 같네요! 
제 경우는 PostContentDto에서 static 메서드로 구현하고, from 메서드에서 `List<PostContentDto> contentDtos` 를 받는 식으로 구현할 것 같습니다 !!


```suggestion
    public static PostListResponse from(Post post, List<PostContentDto> postContentDtos) {
        String thumbnailUrl = extractThumbnailUrl(postContentDtos);
        return new PostListResponse(
                post.getId(),
                post.getTitle(),
                postContentDtos,
                post.getCommentCount(),
                thumbnailUrl,
                post.getCreatedAt()
        );
    }

```